packages/itwin/grouping-mapping-widget/src/widget/components/PropertyTable.tsx

```diff
@@ -0,0 +1,80 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { Table } from "@itwin/itwinui-react";
+import React, { useState } from "react";
+import type { Column } from "react-table";
+import type { CreateTypeFromInterface } from "../utils";
+import { useGroupingMappingApiConfig } from "./context/GroupingApiConfigContext";
+import DeleteModal from "./DeleteModal";
+import { PropertyTableToolbar } from "./PropertyTableToolbar";
+
+export interface PropertyTableItem {
+  propertyName: string;
+  id: string;
+}
+
+export interface PropertyTableProps<T extends PropertyTableItem> {
+  propertyType: string;
+  columns: (handleShowDeleteModal: (value: T) => void) => ReadonlyArray<Column<T>>;
+  data: T[];
+  isLoading: boolean;
+  onClickAdd?: () => void;
+  refreshProperties: () => Promise<void>;
+  deleteProperty: (iModelId: string, accessToken: string, propertyId: string) => Promise<void>;
+
+}
+
+export const PropertyTable = <T extends PropertyTableItem>({
+  propertyType,
+  columns,
+  data,
+  isLoading,
+  onClickAdd,
+  refreshProperties,
+  deleteProperty,
+}: PropertyTableProps<T>) => {
+  const { getAccessToken, iModelId } = useGroupingMappingApiConfig();
+  const [showDeleteModal, setShowDeleteModal] = useState<T | undefined>(undefined);
+
+  const handleDeleteProperty = async () => {
+    const accessToken = await getAccessToken();
+    await deleteProperty(iModelId, accessToken, showDeleteModal?.id ?? "");
+  };
+
+  const handleShowDeleteModal = (property: T) => {
+    setShowDeleteModal(property);
+  };
+
+  const handleCloseDeleteModal = () => {
+    setShowDeleteModal(undefined);
+  };
+
+  return (
+    <>
+      <div>
+        <PropertyTableToolbar
+          propertyType={propertyType}
+          onClickAddProperty={onClickAdd}
+          refreshProperties={refreshProperties}
+          isLoading={isLoading}
+        />
+      </div>
+      <Table<CreateTypeFromInterface<T>>
+        data={isLoading ? [] : data}
+        density='extra-condensed'
+        columns={columns(handleShowDeleteModal)}
```

Comment:
Looks like columns will be recreated for every re-render. Earlier columns were stored in a `useMemo`.

Reply:
`useMemo` is used to memoize a value (like a function's result), while `useCallback` is used to memoize a function itself, so that it is not recreated on each render of the component. So this should not happen if the columns callback is wrapped in `useCallback`.

Reply:
But you are assigning a value to `columns` prop, not a callback. Every re-render this `columns` function will be executed and the return value passed into the `columns` props of `Table` component.

Reply:
Oh true, nice catch. My mistake. I have fixed this now.